app/models/problem_check_tracker.rb

```diff
@@ -40,7 +40,14 @@ def no_problem!(next_run_at: nil)
   end
 
   def check
-    ProblemCheck[identifier]
+    check = ProblemCheck[identifier]
+
+    return check if check.present?
+
+    silence_the_alarm
+    destroy
```

Comment:
If the `ProblemCheck` class is no longer loaded, it indicates that the problem check has been removed or renamed.

Reply:
That's a nitpick, but it's a little bit surprising that a function called "check" also does silencing and destroying, not sure the name is correct.

Reply:
I agree with you. I had the same "feeling" when I was working on this.

I think the source of this feeling is that it's actually error handling. Probably `ProblemCheck[:foo]` should raise an error, and this silencing and destroying should happen in a `rescue` block, but I didn't want to make that change here for fear of breakage.

Just as a vibe check, would it make you less uncomfortable if this was done in a `rescue` block? Or would it feel the same to you?

Reply:
yes that wouldn't change much to me, it's more than I don't expect side effects when calling something named "check". In my mind a check should return true or false, or eventually some custom state, but not much more. Anyways Im fine with the current code, that's just a note if it was easy to change, but no need to go into a rabbit hole for this.